src/azure-cli/azure/cli/command_modules/cdn/custom/custom_afdx.py

```diff
@@ -2,634 +2,696 @@
+              args.enable_private_link.to_serialized_data() is True or
+              'sharedPrivateLinkResource' in existing):
+            existing_private_link_location = None if 'sharedPrivateLinkResource' not in existing or \
+                'privateLinkLocation' not in existing['sharedPrivateLinkResource'] \
+                else existing['sharedPrivateLinkResource']['privateLinkLocation']
+            existing_private_link_resource = None if 'sharedPrivateLinkResource' not in existing or \
+                'privateLink' not in existing['sharedPrivateLinkResource'] \
+                else existing['sharedPrivateLinkResource']['privateLink']['id']
+            existing_private_link_request_message = None if 'sharedPrivateLinkResource' not in existing \
+                or 'requestMessage' not in existing['sharedPrivateLinkResource'] else \
+                existing['sharedPrivateLinkResource']['requestMessage']
+            existing_private_link_sub_resource_type = None if 'sharedPrivateLinkResource' not in existing \
+                or 'groupId' not in existing['sharedPrivateLinkResource'] else \
+                existing['sharedPrivateLinkResource']['groupId']
+            shared_private_link_resource = {
+                'private_link_location': args.private_link_location if has_value(args.private_link_location)
+                else existing_private_link_location,
+                'private_link': {
+                    'id': args.private_link_resource if has_value(args.private_link_resource)
+                    else existing_private_link_resource
+                },
+                'request_message': args.private_link_request_message if has_value(args.private_link_request_message)
+                else existing_private_link_request_message,
+                'group_id': args.private_link_sub_resource_type if has_value(args.private_link_sub_resource_type)
+                else existing_private_link_sub_resource_type
+            }
+
+        args.shared_private_link_resource = shared_private_link_resource
+        args.host_name = args.host_name if args.host_name is not None else existing['hostName']
+        args.http_port = args.http_port if args.http_port is not None else existing['httpPort']
+        args.https_port = args.https_port if args.https_port is not None else existing['httpsPort']
+        args.origin_host_header = args.origin_host_header if args.origin_host_header is not None \
+            else existing['originHostHeader']
+        args.priority = args.priority if args.priority is not None else existing['priority']
+        args.weight = args.weight if args.weight is not None else existing['weight']
+        args.enabled_state = args.enabled_state if args.enabled_state is not None else existing['enabledState']
+        args.enforce_certificate_name_check = \
+            args.enforce_certificate_name_check if args.enforce_certificate_name_check is not None \
+            else existing['enforceCertificateNameCheck']
```

Comment:
It's generally considered a good practice to express conditions in a positive manner whenever possible because it tends to make the code more readable and easier to understand.

Or, using comments to make it clearer.